File: library/src/androidTest/java/com/raizlabs/android/dbflow/test/typeconverter/LocationConverter.java
package com.raizlabs.android.dbflow.test.typeconverter;

import android.location.Location;

import com.raizlabs.android.dbflow.converter.TypeConverter;

/**
 * Description:
 */
@com.raizlabs.android.dbflow.annotation.TypeConverter
public class LocationConverter extends TypeConverter<String, Location> {
    @Override
    public String getDBValue(Location model) {
        return String.format("%1s,%1s", model.getLatitude(), model.getLongitude());
    }

    @Override
    public Location getModelValue(String data) {
        if (data == null) {
            return null;
        }
        String[] values = data.split(",");
        Location location = new Location("");
        location.setLatitude(Double.valueOf(values[0]));
        location.setLongitude(Double.valueOf(values[1]));
        return location;
    }
}

File: dbflow-tests/src/test/java/com/raizlabs/android/dbflow/typeconverter/LocationConverter.kt
package com.raizlabs.android.dbflow.typeconverter

import android.location.Location

import com.raizlabs.android.dbflow.converter.TypeConverter

/**
 * Description:
 */
@com.raizlabs.android.dbflow.annotation.TypeConverter
class LocationConverter : TypeConverter<String, Location>() {

    override fun getDBValue(model: Location?): String? {
        return if (model == null) null else String.format("%1s,%1s", model.latitude, model.longitude)
    }

    override fun getModelValue(data: String?): Location? {
        if (data == null) {
            return null
        }
        val values = data.split(",".toRegex()).dropLastWhile(String::isEmpty).toTypedArray()
        val location = Location("")
        location.latitude = java.lang.Double.valueOf(values[0])!!
        location.longitude = java.lang.Double.valueOf(values[1])!!
        return location
    }
}
